flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java

```diff
@@ -81,6 +84,12 @@
 	/** Reserved memory per memory owner. */
 	private final Map<Object, Map<MemoryType, Long>> reservedMemory;
 
+	/** Reserved closeable shared objects for state backends fetched from this memory manager. */
+	private final AtomicReference<Deque<AutoCloseable>> stateBackendSharedObjects;
```

Comment:
I suggest to introduce a `RocksDBSharedObject` wrapping up `Cache` and `WriteBufferManager` objects to attach here to simplify the logic. Please refer to [this review branch](https://github.com/carp84/flink/tree/flink-14484-review) for detailed idea.

Reply:
Thanks for this idea.